CMakeLists.txt

```diff
@@ -164,6 +169,14 @@ else()
     MESSAGE(FATAL_ERROR "Unknown CRYPTO")
 endif()
 
+if (X509_IGNORE_CRITICAL STREQUAL "ON")
+    if (CRYPTO STREQUAL "openssl")
+        add_definitions(-DOPENSSL_IGNORE_CRITICAL=1)
+    else()
```

Comment:
Should be an `else if` for `mbedtls`.

Reply:
Although the current Libspdm only have two crypto library, there may be other crypto library in the future.
https://github.com/DMTF/libspdm/blob/16bf4380cdb514e0a6d7682955e2f48cc0e1e9eb/CMakeLists.txt#L147-L165

And I have added the check.  Thanks.
https://github.com/DMTF/libspdm/blob/6473dc78def184cfc0bb4fcf72a19c8b500b73f0/CMakeLists.txt#L172-L178